command/validate/validate.go

```diff
@@ -40,7 +45,7 @@ func (c *cmd) Run(args []string) int {
 		c.UI.Error("Must specify at least one config file or directory")
 		return 1
 	}
```

Comment:
This should also validate that `config-format` must be "hcl" or "json" and fail fast returning 1 otherwise, similar to above where it validates the length of config file.

Reply:
Currently the value of config-format is validated here: https://github.com/hashicorp/consul/blob/92ceaaad81b6e36d782439e3fa997dbf472bb756/agent/config/builder.go#L236

Should it be validated at the top level too?

Reply:
That should still say because the builder is also called when you start the agent. 

I suggested adding a validation check here in the CLI code and failing faster so that we can see a nicer error message  when running `consul validate foo.txt` without an extra wrapping layer, it will go from 
`Config validation failed: config: -config-format must be either 'hcl' or 'json'`
to something like 
`-config-format must be either 'hcl' or 'json`

any invalid values _inside_ the file being sent to the builder should get the error message from there, but there's no reason to call the builder here if we know the configFormat is wrong.